frontend/src/components/file/AdvancedFileUpload.vue

```diff
@@ -55,12 +75,18 @@ const onFileUploadDragAndDrop = (event: FileUploadUploaderEvent) => {
 const onUpload = async (files: Array<File>) => {
   uploading.value = true;
 
+  if (!activityId) {
+    curActivityId.value = await generateActivityId();
+  } else {
+    curActivityId.value = activityId;
+  }
```

Comment:
Optional style: You might be able to get away with the following one-liner ternary
```ts
curActivityId.value = activityId ? activityId : await generateActivityId();
```

Reply:
done and noted!